sources/src/main/java/com/google/solutions/jitaccess/core/adapters/ResourceManagerAdapter.java

```diff
@@ -248,6 +248,28 @@ public List<String> testIamPermissions(
     }
   }
 
+  public List<ProjectId> listProjects(String parent) throws NotAuthenticatedException, IOException {
+    try {
+      var response = createClient()
+        .projects()
+        .list()
+        .setParent(parent)
```

Comment:
AFAIK, this only lists the _immediate_ children of that folder/organization. I figure that might be too limiting for organizations that use nested folders.

I am not aware of a way to list all descendants of a folder/organization. But what we can do instead is call [`search`](https://cloud.google.com/resource-manager/reference/rest/v3/projects/search) with a query `state='ACTIVE'`. That returns all projects that the application's service account can "see". If the application has been deployed according to the docs, then this is equivalent to all projects underneath `RESOURCE_SCOPE`.

Wdyt?

Reply:
A good point, let's switch to `search`. I'll update the PR accordingly.

Reply:
Updated the PR with the new implementation `search` instead of `list`.